src/flint/fmpz_mat.jl

```diff
@@ -618,6 +623,58 @@ Reduce the entries of $x$ modulo $y$ and return the result.
 """
 reduce_mod(x::ZZMatrix, y::Integer) = reduce_mod(x, ZZRingElem(y))
 
+@doc raw"""
+    mod!(M::ZZMatrix, p::ZZRingElem)
+
+Reduces every entry modulo $p$ in-place, i.e. applies the mod function to every entry.
+Positive residue system.
+"""
+function mod!(M::ZZMatrix, p::ZZRingElem)
+  GC.@preserve M begin
+    for i = 1:nrows(M)
+      for j = 1:ncols(M)
+        z = ccall((:fmpz_mat_entry, libflint), Ptr{ZZRingElem}, (Ref{ZZMatrix}, Int, Int), M, i - 1, j - 1)
+        ccall((:fmpz_mod, libflint), Nothing, (Ptr{ZZRingElem}, Ptr{ZZRingElem}, Ref{ZZRingElem}), z, z, p)
+      end
+    end
+  end
+  return nothing
+end
+
+@doc raw"""
+    mod(M::ZZMatrix, p::ZZRingElem) -> ZZMatrix
+
+Reduces every entry modulo $p$, i.e. applies the mod function to every entry.
+"""
+function mod(M::ZZMatrix, p::ZZRingElem)
```

Comment:
I note that there is an existing function `reduce_mod` which seems to do the same, albeit with a direct call to function `fmpz_mat_scalar_mod_fmpz` which I am guessing be more efficient?

What should we do, keep both `reduce_mod` and `mod`? Or merge them? Thoughts, @fieker @thofma ?